### PR TITLE
Allow property security-inter-broker-protocol 

### PR DESCRIPTION
**Pull request by cawright-rh**
## Description

Please provide a meaningful description of what this change will do, or is for. Bonus points for including links to
related issues, other PRs, or technical references.

This change is to allow for the kafka property `security.inter.broker.protocol` to be used by the broker pods. Currently, koperator will create the property `inter.broker.listener.name,` which is mutually exclusive with `security.inter.broker.protocol`. 

see kafka documentation on the property: https://kafka.apache.org/documentation/#brokerconfigs_security.inter.broker.protocol 

Note that by _not_ including a description, you are asking reviewers to do extra work to understand the context of this
change, which may lead to your PR taking much longer to review, or result in it not being reviewed at all.

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](../CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [x] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: pkg/resources/kafka/kafka.go [hunk 1]
```diff
@@ -1501,14 +1501,18 @@ func getServiceFromExternalListener(client client.Client, cluster *v1beta1.Kafka
...
+			iControllerServiceName = strings.ReplaceAll(iControllerServiceName, "_", "-")
 		}
 	case envoyutils.IngressControllerName:
 		if ingressConfigName == util.IngressConfigGlobalName {
 			iControllerServiceName = fmt.Sprintf(envoyutils.EnvoyServiceName, eListenerName, cluster.GetName())
+			iControllerServiceName = strings.ReplaceAll(iControllerServiceName, "_", "-")
 		} else {
 			iControllerServiceName = fmt.Sprintf(envoyutils.EnvoyServiceNameWithScope, eListenerName, ingressConfigName, cluster.GetName())
+			iControllerServiceName = strings.ReplaceAll(iControllerServiceName, "_", "-")
 		}
```
Comment (dobrerazvan) [inline pkg/resources/kafka/kafka.go:1516]: Please rebase, I merged the contour work and there is some more code in this switch case.

**Comment by dobrerazvan** [comment]
Looks good to me, just to rebase to have the latest changes.

File: api/v1beta1/kafkacluster_types.go [hunk 2]
```diff
@@ -748,6 +748,7 @@ type CommonListenerSpec struct {
 	// At least one of the listeners should have this flag enabled
 	// +optional
 	UsedForInnerBrokerCommunication bool `json:"usedForInnerBrokerCommunication"`
+	UsedForKafkaAdminCommunication  bool `json:"usedForKafkaAdminCommunication,omitempty"`
```
Comment (dobrerazvan) [inline api/v1beta1/kafkacluster_types.go:751]: Add description for when this flag is useful.

File: pkg/util/client/common.go [hunk 3]
```diff
@@ -37,11 +37,18 @@ func UseSSL(cluster *v1beta1.KafkaCluster) bool {
 
 func getContainerPortForInnerCom(internalListeners []v1beta1.InternalListenerConfig, extListeners []v1beta1.ExternalListenerConfig) int32 {
 	for _, val := range internalListeners {
+		if val.UsedForKafkaAdminCommunication { // Cannot Currently update the CRD as I do not have permissions
```
Comment (dobrerazvan) [inline pkg/util/client/common.go:40]: CRDs are not updated in all envs, is this still needed?

File: pkg/util/client/common.go [hunk 4]
```diff
@@ -37,11 +37,18 @@ func UseSSL(cluster *v1beta1.KafkaCluster) bool {
...
+			return val.ContainerPort
+		}
 		if val.UsedForInnerBrokerCommunication {
```
Comment (dobrerazvan) [inline pkg/util/client/common.go:43]: Dito

File: pkg/resources/kafka/configmap.go [hunk 5]
```diff
@@ -88,8 +88,10 @@ func (r *Reconciler) getConfigProperties(bConfig *v1beta1.BrokerConfig, id int32
 	}
 
 	// Add Cruise Control Metrics Reporter configuration
-	if err := config.Set(kafkautils.CruiseControlConfigMetricsReporters, "com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter"); err != nil {
-		log.Error(err, fmt.Sprintf("setting '%s' in broker configuration resulted an error", kafkautils.CruiseControlConfigMetricsReporters))
+	if !strings.Contains(r.KafkaCluster.Spec.ReadOnlyConfig, kafkautils.KafkaConfigSecurityInterBrokerProtocol+"=") {
```
Comment (azun) [inline pkg/resources/kafka/configmap.go:91]: I think you can convert the readonly config to properties using func getBrokerReadOnlyConfig

File: pkg/util/client/common.go (hunk 3 above)
Comment (cawright-rh) [reply pkg/util/client/common.go:40]: Ah, this comment should be removed, the conditional statement is still needed though, I will add a comment explaining the use. The summary is, having a separate flag to indicate which port to use for the koperator and CruiseControl logic is valuable, as, it normally just uses UsedForInnerBrokerCommunication port, which may not be a port CC or Koperator has access to.

File: pkg/resources/kafka/configmap.go (hunk 5 above)
Comment (azun) [reply pkg/resources/kafka/configmap.go:91]: Also, update the comment to reflect that the reported is disabled when inter broker communication is secured

File: pkg/resources/kafka/configmap.go [hunk 6]
```diff
@@ -292,9 +296,12 @@ func generateListenerSpecificConfig(l *v1beta1.ListenersConfig, serverPasses map
 	if err := config.Set(kafkautils.KafkaConfigListenerSecurityProtocolMap, securityProtocolMapConfig); err != nil {
 		log.Error(err, fmt.Sprintf("setting '%s' parameter in broker configuration resulted an error", kafkautils.KafkaConfigListenerSecurityProtocolMap))
 	}
-	if err := config.Set(kafkautils.KafkaConfigInterBrokerListenerName, interBrokerListenerName); err != nil {
-		log.Error(err, fmt.Sprintf("setting '%s' parameter in broker configuration resulted an error", kafkautils.KafkaConfigInterBrokerListenerName))
+	if !strings.Contains(r, kafkautils.KafkaConfigSecurityInterBrokerProtocol+"=") {
```
Comment (azun) [inline pkg/resources/kafka/configmap.go:299]: This means the inter broker listener needs to be explicitly specified through the readonly config right?

**Review by azun — APPROVED** [review]

File: pkg/resources/kafka/configmap.go (hunk 6 above)
Comment (cawright-rh) [reply pkg/resources/kafka/configmap.go:299]: yes that is correct, it is a kafka property similar, but conflicting with, inter.broker.listener.name and inter.broker.listener.protocol

**Review by dobrerazvan — APPROVED** [review]

File: pkg/resources/kafka/configmap.go (hunk 5 above)
Comment (hvan) [reply pkg/resources/kafka/configmap.go]: Updated to use the getBrokerReadOnlyConfig function and added comment.

**Pull request merged**